### DISCUSSION TITLE
Why is docker-compose downloaded when I have docker WITH compose installed already?

### DISCUSSION THREAD

**Discussion by func0der** [ddev/ddev#6655] [Q&A] [answered]
Hey,

I have been wondering since I started using ddev, why it would install `docker-compose`. `docker-compose` is basically a dead or better migrated product at this point.
It has been replaced with `docker compose` (v2 or whatever you may call it).

I could not find documentation, an issue or similar info about this behaviour. I only know it does download it (not even from where xD).

Can somebody please explain?

**Comment by rfay** [comment, marked as answer]
Hi @func0der - `docker-compose` is by no means a dead product. It's actually a docker plugin that can be run either as `docker compose` (if correctly installed in ~/.docker/cli-plugins) or as `docker-compose` (if in the PATH or if explicitly pathed).  If you look at the project releases, you'll see they're all `docker-compose`. https://github.com/docker/compose/releases

DDEV has used `docker-compose` v2 for a number of years. We currently use v2.29.7.

DDEV downloads its own tested and verified private docker-compose so that you do not have to install one. Unfortunately, docker-compose periodically introduces new bugs and so we don't change the version on it until it has been verified. In the early days of docker-compose v2, nearly every release had a new problem. These days it's not such a serious problem, but since we can't control what people may have installed, we test each one before specifying it for use in DDEV (where the private version in ~/.ddev/bin/docker-compose is used).

We do not use any `docker-compose` except the private one for DDEV, and in fact no `docker-compose` or `docker compose` needs to be installed on your system to use DDEV, because DDEV downloads its own.

You can see the version in use on your DDEV installation with `ddev version`

If we were to use the docker plugin approach (`docker compose`), then everybody would have to install the same compose plugin in their ~/.docker/plugins directory, and we wouldn't be able to control the version at all. And no benefit would be derived.
